provider/src/main/java/com/raygun/raygun4android/rum/RUM.java

```diff
@@ -19,225 +19,136 @@
 import com.raygun.raygun4android.network.RaygunNetworkLogger;
 import com.raygun.raygun4android.services.RUMPostService;
 
-import java.lang.ref.WeakReference;
 import java.text.SimpleDateFormat;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.temporal.ChronoUnit;
 import java.util.Calendar;
 import java.util.TimeZone;
 import java.util.UUID;
-import java.util.concurrent.TimeUnit;
 
-public class RUM implements ActivityLifecycleCallbacks {
-    private static RUM rum;
-    private static WeakReference<Activity> mainActivity;
-    private static WeakReference<Activity> currentActivity;
-    private static WeakReference<Activity> loadingActivity;
-    private static long startTime;
-    private static long lastSeenTime;
-    private static String sessionId;
-    private static RaygunUserInfo currentSessionUser;
+public class RUM {
+    // Singleton instance
+    private static RUM instance;
 
-    private static void attach(Activity mainActivity) {
+    private long lastSeenTime;
+    private String sessionId;
+    private RaygunUserInfo currentSessionUser;
+    private final RUMActivity rumActivity;
```

Comment:
All the "Activity" logic was moved to `RUMActivity`, which also facilitates testing it in `RUMActivityTest`